test/Mono.Linker.Tests.Cases.Expectations/Assertions/TestRunCharacteristics.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Mono.Linker.Tests.Cases.Expectations.Assertions
+{
+	[Flags]
+	public enum TestRunCharacteristics
+	{
+		TargetingNetFramework = 1,
+		TargetingNetCore = 2,
+		UsingIllink = 4,
```

Comment:
Good point.  Technically, it would be nice we could run the tests that are currently using this.

I was able to get these tests running.  It took 2 changes.

1) I hardcoded some hack logic in our `UnityTestCaseCompiler` to tweak the references when we see `illink.dll` appear.  I don't see a clean way to handle this.  I can live with the hack in our test framework.

2) I need to change `Driver.FindStep` in order to get `SubStepDispatcherUsage` passing.  It tries to add a step before `MarkStep`, however, we have a `UnityMarkStep` which inherits from `MarkStep`.  I thought a nice way to deal with this would be to change `FindStep` to look like.

```
		static IStep FindStep (Pipeline pipeline, string name)
		{
			foreach (IStep step in pipeline.GetSteps ()) {
				Type t = step.GetType ();
				while (t.BaseType != null) {
					if (t.Name == name)
						return step;
					t = t.BaseType;
				}
			}

			return null;
		}
```
Are you OK with that?  If so, I can update the PR with the necessary changes.

Alternatively, I can make `FindStep` virtual and then override it in our linker.

Reply:
> Alternatively, I can make FindStep virtual and then override it in our linker.

That's my preference too

Reply:
Done.